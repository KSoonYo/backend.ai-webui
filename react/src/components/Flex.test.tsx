import { render, screen } from "@testing-library/react";
import Flex from "./Flex";
import renderer from "react-test-renderer";

describe("Flex", () => {
  test("default render", async () => {
    const flex = renderer.create(<Flex />);
    expect(flex.toJSON()).toMatchSnapshot();
  });

  test("render with custom props", async () => {
    const customFlex = renderer.create(
      <Flex
        direction="column"
        wrap="wrap-reverse"
        justify="center"
        align="start"
        gap="sm"
        style={{ backgroundColor: "blue" }}
      />
    );
    expect(customFlex.toJSON()).toMatchSnapshot();
  });
});
