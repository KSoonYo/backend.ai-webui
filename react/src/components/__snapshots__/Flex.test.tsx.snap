// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex default render 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "border": "0 solid black",
      "boxSizing": "border-box",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexShrink": 0,
      "flexWrap": "nowrap",
      "gap": 0,
      "justifyContent": "flex-start",
      "listStyle": "none",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "textDecoration": "none",
    }
  }
/>
`;

exports[`Flex render with custom props 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "blue",
      "border": "0 solid black",
      "boxSizing": "border-box",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "flexWrap": "wrap-reverse",
      "gap": 12,
      "justifyContent": "center",
      "listStyle": "none",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "textDecoration": "none",
    }
  }
/>
`;
